components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1315,11 +1315,26 @@ void AdsServiceImpl::MigratePrefsVersion2To3() {
 }
 
 void AdsServiceImpl::MigratePrefsVersion3To4() {
-  auto locale = GetLocale();
-  auto region = ads::Ads::GetRegion(locale);
+  const auto locale = GetLocale();
+  const auto region = ads::Ads::GetRegion(locale);
+
+  // Disable ads for unsupported legacy regions due to a bug where ads were
+  // enabled even if the users region was not supported
+  const std::vector<std::string> legacy_regions = {
```

Comment:
Presumably the bug has proven unfixable/untenable before?

Reply:
The bug was caused by Ads being incorrectly enabled in older versions so for unsupported regions we disable ads, show onboarding to allow the user to decide whether they would like to enable ads